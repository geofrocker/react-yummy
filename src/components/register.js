import React, { Component } from 'react';
import url from './config'
class Register extends Component {

    constructor(){
        super()
        this.state={message:''}
    }
    regUser = (e) =>{
        e.preventDefault();
        fetch(url+'auth/register', {
            method: 'POST',
            headers: {
              'Content-Type': 'application/json',
            },
            body: JSON.stringify({
              name: this.refs.name.value,
              username: this.refs.username.value,
              email: this.refs.email.value,
              password: this.refs.password.value
            })
        }).then((response) => response.json())
        .then((responseJson) => {
            console.log(responseJson);
            if(responseJson.status){
                this.setState({message:responseJson.Message})
                this.refs.name.value=null;
                this.refs.username.value=null;
                this.refs.email.value=null;
                this.refs.password.value=null;
                this.refs.cpassword.value=null;
            }
            this.setState({message:responseJson.Message})
 
        })
        .catch((error) => {
          console.error(error);
        });
        
        
       
    }
    render(){
        return (
        <div className="Register">
            <h1>Register Here</h1>
            {this.state.message
                    ? <div className="alert alert-danger col-xs-12">{this.state.message}</div>
                    : <div></div> 
                }
            <form onSubmit={this.regUser}>
                <div className="jumbotron">
                    <div className="form-group">
                        <input type="text" className="form-control" placeholder="Name" ref="name" required/>
                    </div>
                    <div className="form-group">
                        <input type="text" className="form-control" placeholder="Username" ref="username" required/>
                    </div>
                    <div className="form-group">
                        <input type="email" className="form-control" placeholder="Email" ref="email" required/>
                    </div>
                    <div className="form-group">
                        <input type="password" className="form-control" placeholder="Password" ref="password" required/>
                    </div>
                    <div className="form-group">
                        <input type="password" className="form-control" placeholder="Comfirm Password" ref="cpassword" required/>
                    </div>
                    <input type="submit" className="btn btn-primary pull-right" value="Register"/>

                    <center>
                        <p>or</p>
                        <p><a href="/login">Already a member? Click here to login</a></p>
                    </center>
                </div>
                
            </form>
            
        </div>
        );
    }
}

export default Register