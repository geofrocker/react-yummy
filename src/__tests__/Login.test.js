import 'jsdom-global/register';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Login from '../components/login';

describe('login component', () => {
    const wrapper = mount(<Login/>);
    const preventDefault = jest.fn();
    it('renders the Login class', () => {   
        expect(wrapper.find(".Login")).toHaveLength(1);
    });

    it('renders five div jsx elements', () => {
        expect(wrapper.find("div")).toHaveLength(5);        
    });
    it('renders the sign in form and submit data', () =>{
        wrapper.setState({email:'geofrocker2@gmail.com', password:"psalms"});
        wrapper.find("#login-form").simulate('submit', {preventDefault});
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(preventDefault).toBeCalled();
    });

    it('render the two inputs and hides password', () =>{
        expect(wrapper.find("#password")).toHaveLength(1);
        expect(wrapper.find("#username")).toHaveLength(1);
        wrapper.find("#password").simulate('change', {target: {value: 'psalms'}});
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('make sure the username is valid', () =>{
        wrapper.find("#username").simulate('change', {target: {value: 'geofrocker@gmail.com'}});
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})
