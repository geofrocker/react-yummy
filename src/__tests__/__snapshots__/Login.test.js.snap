// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login component make sure the username is valid 1`] = `
<Login>
  <div
    className="Login"
  >
    <h1>
      Please sign in
    </h1>
    <div />
    <div
      className="jumbotron col-xs-11"
    >
      <form
        id="login-form"
        method="POST"
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <input
            className="form-control"
            id="username"
            onChange={[Function]}
            placeholder="Username"
            required={true}
            type="text"
            value="geofrocker@gmail.com"
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            id="password"
            onChange={[Function]}
            placeholder="Password"
            required={true}
            type="password"
            value="psalms"
          />
        </div>
        <input
          className="btn btn-primary pull-right"
          type="submit"
          value="Login"
        />
        <center>
          <p>
            or
          </p>
          <p>
            <a
              href="/register"
            >
              Click here to register
            </a>
          </p>
        </center>
      </form>
    </div>
  </div>
</Login>
`;

exports[`login component render the two inputs and hides password 1`] = `
<Login>
  <div
    className="Login"
  >
    <h1>
      Please sign in
    </h1>
    <div />
    <div
      className="jumbotron col-xs-11"
    >
      <form
        id="login-form"
        method="POST"
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <input
            className="form-control"
            id="username"
            onChange={[Function]}
            placeholder="Username"
            required={true}
            type="text"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            id="password"
            onChange={[Function]}
            placeholder="Password"
            required={true}
            type="password"
            value="psalms"
          />
        </div>
        <input
          className="btn btn-primary pull-right"
          type="submit"
          value="Login"
        />
        <center>
          <p>
            or
          </p>
          <p>
            <a
              href="/register"
            >
              Click here to register
            </a>
          </p>
        </center>
      </form>
    </div>
  </div>
</Login>
`;

exports[`login component renders the sign in form and submit data 1`] = `
<Login>
  <div
    className="Login"
  >
    <h1>
      Please sign in
    </h1>
    <div />
    <div
      className="jumbotron col-xs-11"
    >
      <form
        id="login-form"
        method="POST"
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <input
            className="form-control"
            id="username"
            onChange={[Function]}
            placeholder="Username"
            required={true}
            type="text"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            id="password"
            onChange={[Function]}
            placeholder="Password"
            required={true}
            type="password"
            value="psalms"
          />
        </div>
        <input
          className="btn btn-primary pull-right"
          type="submit"
          value="Login"
        />
        <center>
          <p>
            or
          </p>
          <p>
            <a
              href="/register"
            >
              Click here to register
            </a>
          </p>
        </center>
      </form>
    </div>
  </div>
</Login>
`;
